server/src/main/java/io/crate/expression/operator/BitwiseAndOperator.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.expression.operator;
+
+import io.crate.data.Input;
+import io.crate.metadata.NodeContext;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.functions.Signature;
+import io.crate.types.DataTypes;
+
+public final class BitwiseAndOperator extends Operator<Object> {
+
+    private final Signature signature;
+    private final Signature boundSignature;
+
+    public static final String NAME = "op_&";
+
+    public BitwiseAndOperator(Signature signature, Signature boundSignature) {
+        this.signature = signature;
+        this.boundSignature = boundSignature;
+    }
+
+    @Override
+    public Signature signature() {
+        return signature;
+    }
+
+    @Override
+    public Signature boundSignature() {
+        return boundSignature;
+    }
+
+    public static void register(OperatorModule module) {
+        module.register(
+            Signature.scalar(
+                NAME,
+                DataTypes.INTEGER.getTypeSignature(),
+                DataTypes.INTEGER.getTypeSignature(),
+                Operator.RETURN_TYPE.getTypeSignature()
+            ), BitwiseAndOperator::new
+        );
+        module.register(
+            Signature.scalar(
+                NAME,
+                DataTypes.LONG.getTypeSignature(),
+                DataTypes.LONG.getTypeSignature(),
+                Operator.RETURN_TYPE.getTypeSignature()
+            ), BitwiseAndOperator::new
+        );
+        module.register(
+            Signature.scalar(
+                NAME,
+                DataTypes.SHORT.getTypeSignature(),
+                DataTypes.SHORT.getTypeSignature(),
+                Operator.RETURN_TYPE.getTypeSignature()
+            ), BitwiseAndOperator::new
+        );
```

Comment:
The function is only tested and documented to work with integer values, any reason to register it for short and long as well?

Reply:
I actually did test it manually with integer/smallint/bigint types but since my unit test was based on existing tests from your codebase that didnt distinguish types specifically I didnt bother either. The doc part I wrote should have been more clear.